src/platform/packages/shared/kbn-unified-metrics-grid/src/components/trace_metrics_grid/index.tsx

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+import { EuiFlexGrid, EuiFlexItem, EuiPanel, euiPaletteColorBlind } from '@elastic/eui';
+import { css } from '@emotion/react';
+import type { TraceIndexes } from '@kbn/discover-utils/src';
+import { useFetch } from '@kbn/unified-histogram';
+import type { ChartSectionProps, UnifiedHistogramInputMessage } from '@kbn/unified-histogram/types';
+import React, { useMemo } from 'react';
+import { Provider } from 'react-redux';
+import { Subject } from 'rxjs';
+import { useEsqlQueryInfo } from '../../hooks';
+import { store } from '../../store';
+import { ErrorRateChart } from './error_rate';
+import { LatencyChart } from './latency';
+import { ThroughputChart } from './throughput';
+
+export const chartPalette = euiPaletteColorBlind({ rotations: 2 });
+
+export type DataSource = 'apm' | 'otel';
+
+function TraceMetricsGrid({
+  requestParams,
+  services,
+  input$: originalInput$,
+  searchSessionId,
+  onBrushEnd,
+  onFilter,
+  abortController,
+  indexes,
+  query,
+  dataSource,
+}: ChartSectionProps & {
+  indexes: TraceIndexes;
+  dataSource: DataSource;
+}) {
+  const esqlQuery = useEsqlQueryInfo({
+    query: query && 'esql' in query ? query.esql : '',
+  });
+
+  const kqlFilters = useMemo(() => {
+    if (query && 'query' in query && query.query) {
```

Comment:
Is there something I'm missing with the `in` usage? We can shorten this to not need the `in` check while still resolving to the same result, but you've included it in the update. I just want to understand if I'm not quite getting the need for this verbose check (which is also the case for line 43). For example this all could be:

```ts
return query?.query && [`KQL("${query.query.replaceAll('"', '\\"')}")`] ?? [];
```

Reply:
`Property 'query' does not exist on type 'Query | AggregateQuery'.`

Reply:
Ahhh, typescript shenanigans. You'd think the optional access would be accepted here.